tests/TestCase/Http/Cookie/CookieTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Http\Cookie;
+
+use Cake\Chronos\Chronos;
+use Cake\Http\Cookie\Cookie;
+use Cake\TestSuite\TestCase;
+
+/**
+ * HTTP cookies test.
+ */
+class CookieTest extends TestCase
+{
+
+    /**
+     * Test invalid cookie name
+     *
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The cookie name `no, this wont, work` contains invalid characters.
+     */
+    public function testValidateNameInvalidChars()
+    {
+        new Cookie('no, this wont, work', '');
+    }
+
+    /**
+     * Test empty cookie name
+     *
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The cookie name cannot be empty.
+     */
+    public function testValidateNameEmptyName()
+    {
+        new Cookie('', '');
+    }
+
+    /**
+     * Test decrypting the cookie
+     *
+     * @return void
+     */
+    public function testDecrypt()
+    {
+        $cookie = new Cookie('cakephp', 'Q2FrZQ==.N2Y1ODQ3ZDAzYzQzY2NkYTBlYTkwMmRkZjFmNGI3Mjk4ZWY5ZmExYTA4YmM2ZThjOWFhZWY1Njc4ZDZlMjE4Y/fhI6zv+siabYg0Cnm2j2P51Sghk7WsVxZr94g5fhmkLJ4ve7j54v9r5/vHSIHtog==');
```

Comment:
Doesn't matter how you turn it around phpcs is complaining about this notation with line breaks because of the fucking silly spaces.